src/cpp/fastdds/xtypes/type_representation/TypeObjectUtils.cpp

```diff
@@ -0,0 +1,2952 @@
+            (type_identifier._d() == TI_PLAIN_ARRAY_SMALL &&
+            type_identifier.array_sdefn().header().equiv_kind() != EK_BOTH) ||
+            (type_identifier._d() == TI_PLAIN_ARRAY_LARGE &&
+            type_identifier.array_ldefn().header().equiv_kind() != EK_BOTH) ||
+            (type_identifier._d() == TI_PLAIN_MAP_SMALL &&
+            type_identifier.map_sdefn().header().equiv_kind() != EK_BOTH) ||
+            (type_identifier._d() == TI_PLAIN_MAP_LARGE &&
+            type_identifier.map_ldefn().header().equiv_kind() != EK_BOTH))
+    {
+        indirect_hash = true;
+    }
+    return indirect_hash;
+}
+
+void TypeObjectUtils::l_bound_consistency(
+        LBound bound)
+{
+    if (bound < 256)
+    {
+        throw InvalidArgumentError("bound parameter must be greater than 255");
+    }
+}
+
+void TypeObjectUtils::l_bound_seq_consistency(
+        const LBoundSeq& bound_seq)
+{
+    if (bound_seq.empty())
+    {
+        throw InvalidArgumentError("array_bound_seq parameter must not be empty");
+    }
+    bool large_dimension = false;
+    for (LBound lbound : bound_seq)
+    {
+        if (INVALID_LBOUND == lbound)
+        {
+            throw InvalidArgumentError("bound must always be greater than 0");
+        }
+        if (lbound > 255)
+        {
+            large_dimension = true;
+        }
+    }
+    if (!large_dimension)
+    {
+        throw InvalidArgumentError("no large bound in array_bound_seq");
+    }
+}
+
+void TypeObjectUtils::member_flag_consistency(
+        MemberFlag member_flags)
+{
+    if (!(member_flags & MemberFlagBits::TRY_CONSTRUCT1 || member_flags & MemberFlagBits::TRY_CONSTRUCT2))
+    {
+        throw InvalidArgumentError("Inconsistent MemberFlag: INVALID TryConstructKind");
+    }
+}
+
+void TypeObjectUtils::struct_member_flag_consistency(
+        StructMemberFlag member_flags)
+{
+    member_flag_consistency(member_flags);
+    if (member_flags & MemberFlagBits::IS_KEY && member_flags & MemberFlagBits::IS_OPTIONAL)
+    {
+        throw InvalidArgumentError("Keyed members cannot be optional");
+    }
+    if (member_flags & MemberFlagBits::IS_KEY && !(member_flags & MemberFlagBits::IS_MUST_UNDERSTAND))
+    {
+        throw InvalidArgumentError("Keyed members must have their \"must understand\" attribute set to true");
+    }
+}
+
+void TypeObjectUtils::type_flag_consistency(
+        TypeFlag type_flag)
+{
+    if ((type_flag & TypeFlagBits::IS_APPENDABLE && type_flag & TypeFlagBits::IS_FINAL) ||
+            (type_flag & TypeFlagBits::IS_APPENDABLE && type_flag & TypeFlagBits::IS_MUTABLE) ||
+            (type_flag & TypeFlagBits::IS_FINAL && type_flag & TypeFlagBits::IS_MUTABLE))
+    {
+        throw InvalidArgumentError("Exactly one extensibility flag must be set");
+    }
+}
+
+void TypeObjectUtils::enumerated_literal_flag_consistency(
+        EnumeratedLiteralFlag enumerated_literal_flag)
+{
+    if ((enumerated_literal_flag & enum_literal_flag_mask) != 0)
```

Comment:
Should the same check be added to CollectionElementFlag, StructMemberFlag, UnionMemberFlag and UnionDiscriminatorFlag?

Reply:
Done in 184e6da3b